spark/src/main/scala/org/apache/comet/serde/strings.scala

```diff
@@ -177,21 +187,16 @@ object CometStringRPad extends CometExpressionSerde[StringRPad] {
 
 object CometStringLPad extends CometExpressionSerde[StringLPad] {
 
-  /**
-   * Convert a Spark expression into a protocol buffer representation that can be passed into
-   * native code.
-   *
-   * @param expr
-   *   The Spark expression.
-   * @param inputs
-   *   The input attributes.
-   * @param binding
-   *   Whether the attributes are bound (this is only relevant in aggregate expressions).
-   * @return
-   *   Protocol buffer representation, or None if the expression could not be converted. In this
-   *   case it is expected that the input expression will have been tagged with reasons why it
-   *   could not be converted.
-   */
+  override def getSupportLevel(expr: StringLPad): SupportLevel = {
+    if (expr.str.isInstanceOf[Literal]) {
+      return Unsupported(Some("Scalar values are not supported for the str argument"))
+    }
+    if (!expr.pad.isInstanceOf[Literal]) {
```

Comment:
I don't know if we'll ever hit this. As far as I can see (in `functions.lpad`), Spark expects the pad argument to be a literal as well.

Reply:
Spark doesn't require pad to be a literal:

```
scala> spark.sql("select a, lpad('foo', 6, a) from t1").show
+---+---------------+
|  a|lpad(foo, 6, a)|
+---+---------------+
|  $|         $$$foo|
|  @|         @@@foo|
+---+---------------+
```

Reply:
```
case class StringRPad(str: Expression, len: Expression, pad: Expression = Literal(" "))
```

Reply:
I suppose this is a good example of the benefit of fuzz testing (which is how this issue was discovered). The fuzzer will generate test cases that most developers would not consider. It does seem unlikely that anyone would want to use a column for the pad value, but I suppose it is possible that someone may have that requirement.